### PR TITLE
feat(local): add Links reading

### PR DESCRIPTION
**Pull request by killian-scalian**
- Add reading part for the links
-  to the read_timeseries methode, add one optional input
- Add unitary test

### CODE REVIEW

File: src/antares/service/local_services/link_local.py [hunk 1]
```diff
@@ -133,3 +137,56 @@ def sort_link_properties_dict(ini_dict: Dict[str, str]) -> Dict[str, str]:
             "filter-year-by-year",
         ]
         return dict(sorted(ini_dict.items(), key=lambda item: dict_order.index(item[0])))
+
+    def get_capacity_direct(
```
Comment (MartinBelthle) [inline src/antares/service/local_services/link_local.py:141]: You should add these 3 methods inside the base service and inside the API one

File: src/antares/service/local_services/link_local.py [hunk 2]
```diff
@@ -133,3 +137,56 @@ def sort_link_properties_dict(ini_dict: Dict[str, str]) -> Dict[str, str]:
...
+        self,
+        area_id: str,
```
Comment (MartinBelthle) [inline src/antares/service/local_services/link_local.py:143]: Also it should be typed area_from, area_to to fit with the existing naming we chose

**Review by MartinBelthle — CHANGES_REQUESTED** [review]
Minor change requested

File: src/antares/service/local_services/link_local.py [hunk 3]
```diff
@@ -133,3 +137,56 @@ def sort_link_properties_dict(ini_dict: Dict[str, str]) -> Dict[str, str]:
...
+        link_clusters = []
+        for element in link_path.iterdir():
+            area_from = element.name
+            for content_area in element.iterdir():
+                if (
+                    content_area.is_file()
+                    and content_area.suffix == ".txt"
+                    and content_area.stem.endswith("parameters")
+                ):
+                    link_clusters.append(
```
Comment (MartinBelthle) [inline src/antares/service/local_services/link_local.py:185]: You didn't do the part on LinkUi and LinkProperties. Not really a big deal if you add a TODO but you can do this inside this PR

**Review by MartinBelthle — CHANGES_REQUESTED** [review]

File: src/antares/service/local_services/link_local.py (hunk 2 above)
Comment (killian-scalian) [reply src/antares/service/local_services/link_local.py]: Fixed in the last commit

File: src/antares/service/local_services/link_local.py (hunk 3 above)
Comment (killian-scalian) [reply src/antares/service/local_services/link_local.py:201]: I did it with the aliases validation

File: src/antares/model/link.py [hunk 4]
```diff
@@ -112,6 +123,12 @@ class LinkUi(DefaultLinkUi):
 
 
 class LinkUiLocal(DefaultLinkUi):
+    link_style: LinkStyle = Field(default=LinkStyle.PLAIN, alias="link-style")
+    link_width: float = Field(default=1, alias="link-width")
+    colorr: int = Field(default=112, alias="colorr")
```
Comment (MartinBelthle) [inline src/antares/model/link.py:128]: Don't need to introduce aliases when it's the same as the python variable

File: src/antares/model/link.py [hunk 5]
```diff
@@ -74,6 +76,15 @@ class LinkProperties(DefaultLinkProperties):
 
 
 class LinkPropertiesLocal(DefaultLinkProperties):
+    hurdles_cost: bool = Field(default=False, alias="hurdles-cost")
```
Comment (MartinBelthle) [inline src/antares/model/link.py:79]: Don't think you should introduce these variables as they are already defined in the DefaultLinkProperties. And you don't use the aliases

**Review by MartinBelthle — CHANGES_REQUESTED** [review]
2 minor changes

File: src/antares/model/link.py (hunk 5 above)
Comment (killian-scalian) [reply src/antares/model/link.py:79]: i removed all these alias. I thought it was useful but no changes when i remove it

File: src/antares/model/link.py (hunk 4 above)
Comment (killian-scalian) [reply src/antares/model/link.py]: Removed next commit

File: tests/antares/services/local_services/test_area.py [hunk 6]
```diff
@@ -1374,3 +1375,57 @@ def test_read_thermals_local(self, local_study_w_thermal):
...
+        for link in links:
+            if link.area_from == "fr":
+                matrix = link.get_capacity_direct()
+                matrix_2 = link.get_capacity_indirect()
+                matrix_3 = link.get_parameters()
+
+                pd.testing.assert_frame_equal(matrix.astype(str), df_initial.astype(str), check_dtype=False)
+                pd.testing.assert_frame_equal(matrix_2.astype(str), df_initial.astype(str), check_dtype=False)
+                pd.testing.assert_frame_equal(matrix_3.astype(str), df_initial.astype(str), check_dtype=False)
+            assert link.ui.link_style.value == "plain"
```
Comment (Sigurd-Borge) [inline tests/antares/services/local_services/test_area.py:1420]: Here you're simply testing the default values for each link, not a big deal but doesn't seem necessary to test for all links unless you think there's a random factor that impacts the writing of the files? The default values already have testing in the creation of links.

Comment (killian-scalian) [reply tests/antares/services/local_services/test_area.py:1420]: Here the dataframes are created inside the unitary test. So i compare what i read with is written at the start. But indeed it is not very useful to assert all these 3 matrix

Comment (Sigurd-Borge) [reply tests/antares/services/local_services/test_area.py:1420]: I meant these:
```python
            assert link.ui.link_style.value == "plain"
            assert link.ui.link_width == 1
            assert link.ui.colorb == 112
            assert link.ui.colorg == 112
            assert link.ui.colorr == 112
            assert not link.properties.hurdles_cost
            assert not link.properties.loop_flow
            assert not link.properties.use_phase_shifter
            assert link.properties.transmission_capacities.value == "enabled"
            assert link.properties.asset_type.value == "ac"
            assert isinstance(link.properties.filter_year_by_year, set)
            assert isinstance(link.properties.filter_synthesis, set)
```

**Review by MartinBelthle — APPROVED** [review]
Okay for me, I don't really care about the test do as you like

**Pull request merged**